bats/send_metrics.bats

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bats
+
+setup() {
+    load $BATS_TEST_DIRNAME/helper/common.bash
+    export PATH=$PATH:~/go/bin
+    export NOMS_VERSION_NEXT=1
+    dolt config --global --unset metrics.disabled
+    cd $BATS_TMPDIR
+    mkdir "dolt-repo-$$"
+    cd "dolt-repo-$$"
+    dolt init
+}
+
+teardown() {
+    rm -rf "$BATS_TMPDIR/dolt-repo-$$"
+}
+
+# How these tests work --
+# The initial dolt command will generate an events file
+# and will kick off a separate process that attempts to send the
+# events in this file to the grpc events server.
+# This attempt will fail, since there is no server running, which will allow
+# the events file to persist.
+# Next, we will run the "send-metrics" command with the "--output" flag
+# which will parse the events file and send it's info to stdout.
+# Because we've implemented a locking system, our "send-metrics" call should
+# fail with a status code of 2 if try to do work on the events directory while
+# a separate process is already working on it
+
+# Create events data then flush data to stdout non-concurrently
+@test "create events data then flush data to stdout non-concurrently" {
+    run dolt ls
+    [ "$status" -eq 0 ]
+    run sleep 1
+    run dolt send-metrics --output
+    [ "$status" -eq 0 ]
+}
+
+# Create events data then flush data to stdout concurrently
+@test "create events data then flush data to stdout concurrently" {
```

Comment:
I feel like this test could be flaky.  I believe it is supposed to do the following:

1. "run dolt ls" creates an event and kicks off a new process to send the event
2. The new process created by the ls command grabs the lock and sends the event
3. "run dolt send-metrics" tries to acquire the lock and fails.
4. "run dolt send-metrics" tries to acquire the lock and fails.

This could fail if 2 & 3 get run in a different order, or if 2 completes before 3 starts.

I think I would do the following 

Prep work:
Create multiple event files in the helper directory and check those in to source control

Then in the test:
1. copy the events to the place where dolt expects them.
2. "run dolt send-metrics --output >file1.txt &"
3. pid1=$!
4. "run dolt send-metrics --output >file2.txt &"
5. pid2=$!
6. wait $pid1
7. exit_code1=$?
8. wait $pid2
9. exit_code2=$?
10. check results.  something like:

if $exit_code1 = 0; then
  if $exit_code2 = 0; then
    # test file1 and file2 for some text like "0 events processed" and make sure the process writes that out
  elif exit_code2 != 2; then
    # fail test
  fi
elif $exit_code2 != 0; then
  # fail test
else
  if exit_code1 != 2; then
    # fail
  fi
fi